### PR TITLE
CSPL-2126: Fixed probe updates

### PR DESCRIPTION
**Pull request by pdhanoya-splunk**
* Added logic to update the pods with latest changes for probe values
* Updates probes example in docs

### CODE REVIEW

File: docs/HealthCheck.md [hunk 1]
```diff
@@ -27,12 +27,12 @@ metadata:
 spec:
   replicas: 1
   startupProbe:
-    initialDelaySeconds: 10
-    timeoutSeconds: 5
-    periodSeconds: 5
-    failureThreshold: 16
+    initialDelaySeconds: 40
```
Comment (akondur) [inline docs/HealthCheck.md:30]: Nit: Are these default values?

File: pkg/splunk/controller/util.go [hunk 2]
```diff
@@ -211,15 +211,30 @@ func MergePodSpecUpdates(ctx context.Context, current *corev1.PodSpec, revised *
 				result = true
 			}
 
-			if current.Containers[idx].LivenessProbe != nil && revised.Containers[idx].LivenessProbe != nil &&
-				current.Containers[idx].LivenessProbe.InitialDelaySeconds != revised.Containers[idx].LivenessProbe.InitialDelaySeconds {
-				current.Containers[idx].LivenessProbe.InitialDelaySeconds = revised.Containers[idx].LivenessProbe.InitialDelaySeconds
+			/*
```
Comment (akondur) [inline pkg/splunk/controller/util.go:214]: Nit: Comments not needed? Apologies if you already were planning to remove it.

**Review by akondur — APPROVED** [review]

File: docs/HealthCheck.md (hunk 1 above)
Comment (pdhanoya-splunk) [reply docs/HealthCheck.md:30]: Yes. Changed them to default values.

File: pkg/splunk/controller/util.go (hunk 2 above)
Comment (pdhanoya-splunk) [reply pkg/splunk/controller/util.go:214]: Thanks for reminder. Will update.

File: pkg/splunk/controller/util.go [hunk 3]
```diff
@@ -308,3 +311,20 @@ func MergeServiceSpecUpdates(ctx context.Context, current *corev1.ServiceSpec, r
 
 	return result
 }
+
+// compareProbeValues compares the current and revised probe values
+func compareProbeValues(currentProbe *corev1.Probe, revisedProbe *corev1.Probe) bool {
```
Comment (gaurav-splunk) [inline pkg/splunk/controller/util.go:316]: nit: the name of the function does not match what it is doing. Maybe something like `hasProbesChanged` or something like that..?

Comment (sgontla) [inline pkg/splunk/controller/util.go:316]: nit: You may want to rename it as `hasProbeChanged` or something like that.

**Review by sgontla — APPROVED** [review]

**Review by kumarajeet — APPROVED** [review]

**Review by gaurav-splunk — APPROVED** [review]

**Pull request merged**